src/m365/yammer/commands/yammer-search.ts

```diff
@@ -0,0 +1,214 @@
+import { Logger } from '../../../cli';
+import { CommandOption } from '../../../Command';
+import GlobalOptions from '../../../GlobalOptions';
+import request from '../../../request';
+import YammerCommand from '../../base/YammerCommand';
+import commands from '../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  search: string;
+  limit?: number;
+}
+
+class YammerSearchCommand extends YammerCommand {
+  private summary: any;
+  private messages: any[];
+  private groups: any[];
+  private topics: any[];
+  private users: any[];
+
+  constructor() {
+    super();
+    this.summary = {
+      messages: 0,
+      groups: 0,
+      topics: 0,
+      users: 0
+    }
+    this.messages = [];
+    this.groups = [];
+    this.topics = [];
+    this.users = [];
+  }
+
+  public get name(): string {
+    return `${commands.YAMMER_SEARCH}`;
+  }
+
+  public get description(): string {
+    return 'Returns a list of messages, users, topics and groups that match the user’s search query.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.search = typeof args.options.search !== 'undefined';
```

Comment:
Since `search` is required, there is no point in tracking it in telemetry as it would always show up as set